perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -184,7 +199,15 @@ public void setUserExtSourceAttributes(PerunSession session, UserExtSource ues,
 							attributeWithValue = attributesManagerBl.getAttribute(session, ues, attr.getName());
 						}
 					}
-					attributeWithValue.setValue(attrValue);
+
+					//for Array list attributes try to parse string value into individual fields
+					if(attributeWithValue.getType().equals(BeansUtils.largeArrayListClassName)) {
```

Comment:
Please check both types, ArrayList and LargeArrayList.

Reply:
Heh, I didn't understand that, hopefully fixed now. :)